### PR TITLE
added gcloud auth

### PR DESCRIPTION
**Pull request by hjfbynara**
This PR adds a step to grab google SA credentials from vault and authing as SA.  The vault step is utilizing App-Role authentication (modeled from Jade/Monster teams).  The SA is a new sa that only has access to non-prod environments.

I unfortunately don't know of a way to test the git action without actually put it in place.

### CODE REVIEW

File: .github/workflows/TerraformDeployTest-gotc-dev.yml [hunk 1]
```diff
@@ -53,15 +84,13 @@ jobs:
...
-#        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    - name: Terraform plan
+      uses: hashicorp/terraform-github-actions@master
+      with:
+        tf_actions_version: ${{ env.terraform_version }}
+        tf_actions_subcommand: 'plan'
+        tf_actions_comment: true
+        tf_actions_working_dir: ${{ env.terraform_directory }}
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (rexwangcc) [inline .github/workflows/TerraformDeployTest-gotc-dev.yml]: Which token are we using here in the Github Settings secrets? I've seen a lot of Actions that require `personal token` (the Helm Chart Releaser needs this too), it'd be great have a shared machine user and the token shared across the Actions.

File: .github/workflows/TerraformDeployTest-gotc-dev.yml [hunk 2]
```diff
@@ -30,6 +30,37 @@ jobs:
...
+           exit 1
+        fi
+        curl --silent -H "X-Vault-Token: ${VAULT_TOKEN}" -X GET \
+             ${VAULT_ADDR}/v1/secret/dsde/gotc/dev/ci/ci-non-prod-service-account.json \
+             | jq .data > sa.json
+        if [ ! -s sa.json ] ; then
+           echo "Retrieval of Gcloud SA credentials failed" 
+           exit 1
+        fi
+        gcloud auth activate-service-account --key-file=sa.json
```
Comment (rexwangcc) [inline .github/workflows/TerraformDeployTest-gotc-dev.yml]: How does this work? Don't see any place that installs gcloud (e.g. something like setup-gcloud action) 🤔

File: .github/workflows/TerraformDeployTest-gotc-dev.yml (hunk 1 above)
Comment (hjfbynara) [reply .github/workflows/TerraformDeployTest-gotc-dev.yml]: Github actions creates a token automatically when you create your first action.   This token is hidden and is used by the github action runners in order for it to access your repo.  It is a token just for this repo and has no access to anything beyond this repo.  So there is no need to create a personal access token or deployment key.   Here is a URL reference: https://help.github.com/en/actions/configuring-and-managing-workflows/authenticating-with-the-github_token

File: .github/workflows/TerraformDeployTest-gotc-dev.yml (hunk 2 above)
Comment (hjfbynara) [reply .github/workflows/TerraformDeployTest-gotc-dev.yml]: github action runners have a rich set of tools already installed.  Here is a list of what is installed on ubuntu:latest (specified by using the line:     runs-on: ubuntu-latest in the action)

https://github.com/actions/virtual-environments/blob/master/images/linux/Ubuntu1804-README.md

File: .github/workflows/TerraformDeployTest-gotc-dev.yml (hunk 1 above)
Comment (rexwangcc) [reply .github/workflows/TerraformDeployTest-gotc-dev.yml]: Yeah, that's what the official docs said. However, a lot of actions have reported a bug with the auto-generated token with Github Actions: https://github.com/helm/chart-releaser-action#pre-requisites

File: .github/workflows/TerraformDeployTest-gotc-dev.yml (hunk 2 above)
Comment (rexwangcc) [reply .github/workflows/TerraformDeployTest-gotc-dev.yml]: I see, TIL, thanks!

File: .github/workflows/TerraformDeployTest-gotc-dev.yml [hunk 3]
```diff
@@ -56,10 +60,13 @@ jobs:
            echo "Gcloud auth failed!"
            exit 1
         fi
-        env:
```
Comment (rexwangcc) [inline .github/workflows/TerraformDeployTest-gotc-dev.yml]: Oh weird, did not notice this earlier :( Good catch!

**Pull request merged**